utils/lazy-pages-fuzzer/src/config.rs

```diff
@@ -0,0 +1,82 @@
+// This file is part of Gear.
+
+// Copyright (C) 2021-2024 Gear Technologies Inc.
+// SPDX-License-Identifier: GPL-3.0-or-later WITH Classpath-exception-2.0
+
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see <https://www.gnu.org/licenses/>.
+
+use std::num::{NonZeroU32, NonZeroUsize};
+use wasm_smith::InstructionKind;
+
+use gear_wasm_gen::{
+    ConfigsBundle, GearWasmGeneratorConfig, MemoryPagesConfig, SelectableParams,
+    SyscallsConfigBuilder, SyscallsInjectionTypes,
+};
+use gear_wasm_instrument::{
+    gas_metering::MemoryGrowCost, parity_wasm::elements::Instruction, Rules,
+};
+
+use crate::{
+    generate::{InjectGlobalsConfig, InjectMemoryAccessesConfig},
+    INITIAL_PAGES,
+};
+
+#[derive(Debug, Default, Clone)]
+pub struct FuzzerConfigBundle {
+    pub memory_accesses: InjectMemoryAccessesConfig,
+    pub globals: InjectGlobalsConfig,
+}
+
+impl ConfigsBundle for FuzzerConfigBundle {
+    fn into_parts(self) -> (GearWasmGeneratorConfig, SelectableParams) {
+        use InstructionKind::*;
+        (
+            GearWasmGeneratorConfig {
+                memory_config: MemoryPagesConfig {
+                    initial_size: INITIAL_PAGES,
+                    upper_limit: None,
+                    stack_end_page: None,
+                },
+                syscalls_config: SyscallsConfigBuilder::new(SyscallsInjectionTypes::all_never())
+                    .build(),
+                remove_recursions: true,
+                ..Default::default()
+            },
+            SelectableParams {
+                allowed_instructions: vec![
+                    Numeric, Reference, Parametric, Variable, Table, Memory, Control,
+                ],
+                max_instructions: 500,
+                min_funcs: NonZeroUsize::new(3).expect("non zero value"),
+                max_funcs: NonZeroUsize::new(5).expect("non zero value"),
+            },
+        )
+    }
+}
+
+pub struct FuzzerCostRules;
+
+impl Rules for FuzzerCostRules {
```

Comment:
I guess using arbitrary instructions costs and etc. doesn't affect the fuzzing quality for the current fuzzer. Please, @grishasobol double check

Reply:
Cost, instrumentation doesn't really matter for this fuzzer, it's only added to prevent endless loops.